lib/Sema/ConstraintSystem.cpp

```diff
@@ -513,6 +513,12 @@ ConstraintLocator *ConstraintSystem::getCalleeLocator(
                                   {LocatorPathElt::ApplyFunction(),
                                    LocatorPathElt::ImplicitCallAsFunction()});
     }
+
+    // Handling an apply for a nominal type that supports @dynamicCallable.
+    if (fnTy->hasDynamicCallableAttribute()) {
```

Comment:
Ah I guess that can't really be done because generic types are opened separately. Nevermind...

Reply:
> Ah I guess that can't really be done because generic types are opened separately. Nevermind...

Ahh thank's @xedin, just unresolved to ask one thing ... Opened generic types is a thing that I don't fully understand yet, there is a brief overview on TypeChecker.rst but if you can point out some doc or any specific part of the code base where one can find more details will be really appreciated :)

Reply:
It all boils down to `openGeneric*` calls in `ConstraintSystem`. When solver encounters a generic type it tries to "open" it by replacing generic parameters with type variables and translating any requirements into constraints, this is done to make it easy to infer types for a generic parameter and test them against their requirements e.g.

`func foo<T: BinaryInteger>(_: T) -> T { ... }`

when `foo` is referenced it's going to be opened into `foo<$T>(_: $T) -> $T` with additional constraint - `$T conforms to BinaryInteger`.

Reply:
Humm got it! Thank you for the explanation @xedin :)

Reply:
No problem!